src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java

```diff
@@ -0,0 +1,106 @@
+import com.exadel.project.trainee.repository.TraineeRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.doReturn;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+//TODO test additionalInfoService
+@SpringBootTest
+@AutoConfigureMockMvc
+@DisplayName("CRUD tests with Trainee's Entity")
+public class TraineeCrudTest {
+
+    @MockBean
+    private AdditionalInfoRepository additionalInfoRepository;
+
+    @MockBean
+    private InterviewPeriodRepository interviewPeriodRepository;
+
+    @MockBean
+    private CountryRepository countryRepository;
+
+    @MockBean
+    private TraineeRepository traineeRepository;
+
+    @MockBean
+    private InternshipService internshipService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private TraineeTestData traineeTestData;
+
+    @Nested
+    @DisplayName("Check Trainee's create method")
+    class testCreateTraineeMethod{
+
+        @Test
+        @DisplayName("When Trainee doesn't exist in database Then save and return him ")
+        void testCreateTrainee() throws Exception{
+            TraineeDTO traineeDTO = traineeTestData.getTestTraineeDto();
+            TraineeDTO savedTraineeDto = traineeTestData.getTestTraineeDto();
+            savedTraineeDto.setId(null);
+            savedTraineeDto.setTraineeStatus(null);
+            Country belarus = traineeTestData.getTestCountry();
+            Trainee trainee = traineeTestData.getTestTrainee();
+            Trainee savedTrainee = traineeTestData.getTestTrainee();
+            savedTrainee.setId(null);
+            InterviewPeriod interviewPeriod = traineeTestData.getTestInterviewPeriod();
+            InterviewPeriod savedInterviewPeriod = traineeTestData.getTestInterviewPeriod();
+            savedInterviewPeriod.setId(null);
+            AdditionalInfo additionalInfo = traineeTestData.getTestAdditionalInfo();
+            Internship internship = traineeTestData.getTestInternship();
+
+            doReturn(null).when(traineeRepository).findTraineeByEmail(anyString());
```

Comment:
When we want to create an trainee, we must check if we have created an trainee with the same email. can we pass traineeDTO.getEmail to the find method?

Reply:
thanks, fixed